### PR TITLE
Horizontally scrolling tables

### PR DESCRIPTION
**Pull request by Stanton**
table

Should be available in the 'Horizontal Table' tab on http://localhost:3000/app/app.php/lexicon/tables.html.twig

## Browser testing

- [x] Chrome
- [x] Safari
- [x] Firefox
- [x] Edge
- [x] IE11
- [x] IE10
- [x] IE9
- [x] IE8
- [x] Mobile Safari
- [x] Samsung Internet

### CODE REVIEW

**Comment by Stanton** [comment]
@nvasileiadis Do you want to check this branch out and have a go at reviewing the changes for any obvious issues? Good opportunity to get some Windows VMs installed from https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/ and take it for a cross browser testing spin for the desktop browsers?

**Comment by nvasileiadis** [comment]
@Stanton - Windows 8.1 and IE11 the mobile version of the menu isn't working properly and I get this http://prntscr.com/drv7fm . A horizontal scrollbar. A top left "menu" button was expected.

**Comment by Stanton** [comment]
Good spot, I'll see if I can replicate this locally this evening.

**Comment by Stanton** [comment]
@nvasileiadis I'm able to replicate your issue and it's also present on the develop branch so I've raised a separate ticket #448. If you're able to continue testing this one while ignoring the navigation issue, that'd be great 👍

**Comment by nvasileiadis** [comment]
column-ordering-issue

In Horizontal Scroll Tab there is an ordering issue when the user changes selection of column to order.

**Comment by Stanton** [comment]
Blocking this issue until #453 is resolved, as they're quite closely related.

File: js/PulsarUIComponent.js [hunk 1]
```diff
@@ -105,6 +109,54 @@ PulsarUIComponent.prototype.initDataTables = function () {
...
+
+    this.$html.find('.table--horizontal').each(function() {
+        var $table = $(this).parent();
+
+        $table.scroll(function() {
+            component.styleTableOverflows($table);
+        });
+
+        $($window).on('load resize', function () {
+            component.styleTableOverflows($table);
```
Comment (jamesjacobs) [inline js/PulsarUIComponent.js]: $($window)?

File: js/PulsarUIComponent.js [hunk 2]
```diff
@@ -105,6 +110,54 @@ PulsarUIComponent.prototype.initDataTables = function () {
...
+        });
+    });
+};
+
+PulsarUIComponent.prototype.styleTableOverflows = function ($container) {
+
+    var $table = $container.find('.table'),
+        tableFullWidth = $table[0].scrollWidth,
+        tableVisibleWidth = $container.width();
+        console.log(tableFullWidth);
```
Comment (jamesjacobs) [inline js/PulsarUIComponent.js]: can we remove logs now?

File: js/PulsarUIComponent.js [hunk 3]
```diff
@@ -105,6 +110,54 @@ PulsarUIComponent.prototype.initDataTables = function () {
...
+        console.log(tableVisibleWidth);
```
Comment (jamesjacobs) [inline js/PulsarUIComponent.js]: Loggy McLogison

File: js/PulsarUIComponent.js [hunk 4]
```diff
@@ -105,6 +110,54 @@ PulsarUIComponent.prototype.initDataTables = function () {
...
+
+    // Toggle right hand shadow, if overflowing to the right
+    if (tableFullWidth === tableVisibleWidth) {
+        console.log('a');
```
Comment (jamesjacobs) [inline js/PulsarUIComponent.js]: Logs

File: js/PulsarUIComponent.js [hunk 5]
```diff
@@ -105,6 +110,54 @@ PulsarUIComponent.prototype.initDataTables = function () {
...
+        $container
+            .removeClass('table--overflow-right');
+    }
+    else {
+        console.log('b');
```
Comment (jamesjacobs) [inline js/PulsarUIComponent.js]: Logs

File: js/PulsarUIComponent.js [hunk 6]
```diff
@@ -105,6 +110,54 @@ PulsarUIComponent.prototype.initDataTables = function () {
...
+        $container.addClass('table--overflow-right');
+    }
+
+    // Toggle left hand shadow, if overflowing to the left
+    if (($table.offsetParent().offset().left - $table.offset().left) > 0) {
+        console.log('c');
```
Comment (jamesjacobs) [inline js/PulsarUIComponent.js]: Logs

File: js/PulsarUIComponent.js [hunk 7]
```diff
@@ -105,6 +110,54 @@ PulsarUIComponent.prototype.initDataTables = function () {
...
+        $container.addClass('table--overflow-left');
+    }
+    else {
+        console.log('d');
```
Comment (jamesjacobs) [inline js/PulsarUIComponent.js]: Logs

File: js/PulsarUIComponent.js [hunk 8]
```diff
@@ -105,6 +110,54 @@ PulsarUIComponent.prototype.initDataTables = function () {
...
+        $container.removeClass('table--overflow-left');
+    }
+
+    // Remove right hand shadow if table scrolled to right hand edge
+    if (-Math.abs((tableFullWidth - tableVisibleWidth - $table.offsetParent().offset().left)) >= $table.offset().left) {
+        console.log('e');
```
Comment (jamesjacobs) [inline js/PulsarUIComponent.js]: Logs

File: stylesheets/_component.filterbar.scss [hunk 9]
```diff
@@ -126,12 +124,6 @@
 .filter-bar__list {
     line-height: 2em;
     list-style: none;
-
-    .form__control {
```
Comment (jamesjacobs) [inline stylesheets/_component.filterbar.scss]: Why are there filter bar changes? this will need testing in CXM.

File: .travis.yml [hunk 10]
```diff
@@ -20,8 +20,6 @@ install:
 
 script:
   - phpunit
-  - npm test
-  - npm run coverage
```
Comment (mikedevelops) [inline .travis.yml]: Is this removing the tests from the CI build?

File: js/PulsarUIComponent.js [hunk 11]
```diff
@@ -1,16 +1,20 @@
-var $             = require('jquery'),
-    dt            = require('datatables.net')(window, $),
-    dt_buttons    = require('datatables.net-buttons')(window, $),
-    dt_responsive = require('datatables.net-responsive')(window, $),
-    dt_select     = require('datatables.net-select')(window, $),
-    countdown     = require('../libs/jquery.countdown/dist/jquery.countdown.min');
+'use strict';
+
+var $ = require('jquery'),
+    $window = window;
```
Comment (mikedevelops) [inline js/PulsarUIComponent.js]: JJ has raised this below, as we get `$($window)` when we jQuerify it, do we need to store a reference to window at all? I think storing a ref to the window as a jQuery object makes sense, but not of the window itself, we get that for free in the global scope.

File: js/PulsarUIComponent.js (hunk 2 above)
Comment (mikedevelops) [inline js/PulsarUIComponent.js]: I think the hound should be kicking off about this too, if it isn't already.

File: package.json [hunk 12]
```diff
@@ -62,16 +62,15 @@
     "minifyify": "^6.3.0",
     "mocha": "^2.4.5",
     "mocha-phantomjs": "^4.0.2",
-    "mochify": "~2.18.1",
+    "mochify": "~2.15.0",
     "mochify-istanbul": "^2.4.1",
     "node-sass": "^3.7.0",
-    "phantomjs": "2.1.1",
+    "phantomjs": "^1.9.19",
```
Comment (mikedevelops) [inline package.json]: what is the reason for the downgrade?

**Comment by Stanton** [comment]
Looking at the comments I highly suspect a bad merge, especially since this feature was started about 6 months ago 🙈

File: js/PulsarUIComponent.js [hunk 13]
```diff
@@ -105,6 +109,54 @@ PulsarUIComponent.prototype.initDataTables = function () {
     this.$html.find('a[data-toggle="tab"]').on('shown.bs.tab', function (e) {
         $($.fn.dataTable.tables(true)).DataTable().columns.adjust().responsive.recalc();
     });
+
+    this.$html.find('.table--horizontal').each(function() {
+        var $table = $(this).parent();
+
+        $table.scroll(function() {
+            component.styleTableOverflows($table);
```
Comment (Stanton) [inline js/PulsarUIComponent.js]: but it is? or is this because we're in a loop?

**Comment by Stanton** [comment]
@jamesjacobs I've replaced the entire filterbar scss with the version in develop, and resolved the other comments you & @mikedevelops added.

Back to you 🎾

**Comment by Stanton** [comment]
@jamesjacobs Need 👀 over when you get chance

File: js/PulsarUIComponent.js [hunk 14]
```diff
@@ -48,8 +48,8 @@ PulsarUIComponent.prototype.initTables = function () {
 };
 
 PulsarUIComponent.prototype.initDataTables = function () {
-
-    var datatables = this.$html.find('.datatable');
+    var component = this,
+        datatables = this.$html.find('.datatable');
```
Comment (jamesjacobs) [inline js/PulsarUIComponent.js]: Shouldn't `this.$html.find` be `component.$html.find` here?

File: stylesheets/_component.tables.scss [hunk 15]
```diff
@@ -121,6 +121,7 @@
 .table-container {
     margin-bottom: 24px;
     overflow-x: auto;
+    position: relative;
```
Comment (jamesjacobs) [inline stylesheets/_component.tables.scss]: Does this need adding here, if it's just needed for the horizontal changes could it live under `.table--horizontal` so as not to affect other tables?

Comment (Stanton) [inline stylesheets/_component.tables.scss]: It's a parent container, so no.

```
<div class="table-container">
    <table class="table table--horizontal">
        ...
```

**Comment by Stanton** [comment]
[Horizontally scrolling tables](https://trello.com/c/CibgNIKw/41-horizontally-scrolling-tables)

**Pull request merged**